rest/perf_routes.go

```diff
@@ -161,6 +161,60 @@ func (h *perfGetByTaskIdHandler) Run(ctx context.Context) gimlet.Responder {
 	return gimlet.NewJSONResponse(perfResults)
 }
 
+///////////////////////////////////////////////////////////////////////////////
+//
+// GET /perf/task_id/{task_id}/exists
+
+type perfExistsByTaskIdHandler struct {
+	opts data.PerformanceOptions
+	sc   data.Connector
+}
+
+func makeExistsPerfByTaskId(sc data.Connector) gimlet.RouteHandler {
+	return &perfExistsByTaskIdHandler{
+		sc: sc,
+	}
+}
+
+// Factory returns a pointer to a new perfGetByTaskIdHandler.
+func (h *perfExistsByTaskIdHandler) Factory() gimlet.RouteHandler {
+	return &perfExistsByTaskIdHandler{
+		sc: h.sc,
+	}
+}
+
+// Parse fetches the task_id from the http request.
+func (h *perfExistsByTaskIdHandler) Parse(_ context.Context, r *http.Request) error {
+	h.opts.TaskID = gimlet.GetVars(r)["task_id"]
+	h.opts.Tags = r.URL.Query()[tags]
+	return nil
+}
+
+// Run calls the data FindPerformanceResults function and returns the
+// PerformanceResults from the provider.
+func (h *perfExistsByTaskIdHandler) Run(ctx context.Context) gimlet.Responder {
+	perfResults, err := h.sc.FindPerformanceResults(ctx, h.opts)
+	if err != nil {
+		err = errors.Wrapf(err, "problem getting performance results by task id '%s'", h.opts.TaskID)
+		grip.Error(message.WrapError(err, message.Fields{
+			"request": gimlet.GetRequestID(ctx),
+			"method":  "GET",
+			"route":   "/perf/task_id/{task_id}/exists",
+			"task_id": h.opts.TaskID,
+		}))
+		return gimlet.MakeJSONErrorResponder(err)
+	}
+	if len(perfResults) > 0 {
+		existResult := model.APIPerformanceResultExists{
+			Name:   h.opts.TaskID,
```

Comment:
I don't think this is necessarily wrong, but why does this have to return the task ID when the caller already includes the task ID in the route parameter? It just seems superfluous.

Reply:
I'll return the number of results instead